src/ucp/api/ucp.h

```diff
@@ -585,6 +585,77 @@ ucs_status_ptr_t ucp_tag_recv_nb(ucp_worker_h worker, void *buffer, size_t count
 
 /**
  * @ingroup UCP_COMM
+ * @brief Non-blocking probe.
+ *
+ *  Probe for incoming unexpected message. The function returns immediately with
+ * the status.
+ *
+ * @param [in]  worker      UCP worker.
+ * @param [in]  tag         Message tag to expect.
+ * @param [in]  tag_mask    Mask of which bits to match from the incoming tag
+ *                           against the expected tag.
+ * @param [out] info        Filled with details about the received message.
+ *
+ * @return UCS_OK              - A match was found, more details in 'info'.
+ *         UCS_ERR_NO_MESSAGE) - No match found.
+ *         otherwise           - Error during receive.
+ */
+ucs_status_t ucp_tag_probe_nb(ucp_worker_h worker, ucp_tag_t tag,
+                              ucp_tag_t tag_mask, ucp_tag_recv_info_t *info);
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Non-blocking probe and return a message.
+ *
+ *  Probe for incoming unexpected message. The function returns immediately and
+ * possibly returns a handle to the message.
+ *
+ * @param [in]  worker      UCP worker.
+ * @param [in]  tag         Message tag to expect.
+ * @param [in]  tag_mask    Mask of which bits to match from the incoming tag
+ *                           against the expected tag.
+ * @param [out] info        Filled with details about the received message.
+ *
+ * @return UCS_PTR_IS_ERR(UCS_ERR_NO_MESSAGE) - No match found.
+ *         other UCS_PTR_IS_ERR(_ptr)         - Error during receive.
+ *         otherwise                          - A message handle. The handle should
+ *                                              be passed to ucp_tag_msg_recv_nb().
+ */
+ucs_status_ptr_t ucp_tag_msg_probe_nb(ucp_worker_h worker, ucp_tag_t tag,
+                                      ucp_tag_t tag_mask,
+                                      ucp_tag_recv_info_t *info);
+
```

Comment:
Yes, the "_msg" is needed to implement MPI_Mprobe. It extracts the probed message and returns a handle for it, whereas the other probe does not.